tudatBundle/pagmo2/include/pagmo/algorithms/giACOmo.hpp

```diff
@@ -75,44 +75,10 @@ class gi_aco_mo
     /**
-    * Constructs the ACO user defined algorithm for single-objective optimization.
-    *
-    * @param[in] gen Generations: number of generations to evolve.
-    * @param[in] acc Accuracy parameter: for inequality and equality constraints .
-    * @param[in] fstop Objective stopping criterion: when the objective value reaches this value, the algorithm is stopped [for multi-objective, this applies to the first obj. only].
-    * @param[in] impstop Improvement stopping criterion: if a positive integer is assigned here, the algorithm will count the runs without improvements, if this number will exceed IMPSTOP value, the algorithm will be stopped.
-    * @param[in] evalstop Evaluation stopping criterion: same as previous one, but with function evaluations
-    * @param[in] focus Focus parameter: this parameter makes the search for the optimum greedier and more focused on local improvements (the higher the greedier). If the value is very high, the search is more focused around the current best solutions
-    * @param[in] ker Kernel: number of solutions stored in the solution archive
-    * @param[in] oracle Oracle parameter: this is the oracle parameter used in the penalty method
-    * @param seed seed used by the internal random number generator (default is random)
-    * @throws std::invalid_argument if \p acc is not \f$ \in [0,1[\f$, \p fstop is not positive, \p impstop is not a
-    * positive integer, \p evalstop is not a positive integer, \p focus is not \f$ \in [0,1[\f$, \p ants is not a positive integer,
-    * \p ker is not a positive integer, \p oracle is not positive, \p paretomax is not a positive integer, \p epsilon is not \f$ \in [0,1[\f$
-    */
-    gi_aco_mo(unsigned gen = 1u, double acc = 0.95, double  fstop = 1, unsigned impstop = 1, unsigned evalstop = 1,
-          double focus = 0.9,  unsigned ker = 10, double oracle=1.0, unsigned seed = pagmo::random_device::next())
-        : m_gen(gen), m_acc(acc), m_fstop(fstop), m_impstop(impstop), m_evalstop(evalstop), m_focus(focus),
-          m_ker(ker), m_oracle(oracle), m_e(seed), m_seed(seed), m_verbosity(0u),
-          m_log()
-    {
-        if (acc >= 1. || acc < 0.) {
-            pagmo_throw(std::invalid_argument, "The accuracy parameter must be in the [0,1[ range, while a value of "
-                                                   + std::to_string(acc) + " was detected");
-        }
-        if (focus >= 1. || focus < 0.) {
-            pagmo_throw(std::invalid_argument,
-                        "The focus parameter must be in the [0,1[ range, while a value of "
-                            + std::to_string(focus) + " was detected");
-        }
-        if (oracle < 0.) {
-            pagmo_throw(std::invalid_argument,
-                        "The oracle parameter must be >=0, while a value of "
-                            + std::to_string(oracle) + " was detected");
-        }
+    * Empty constructor
+    * */
 
-
-    }
+    //gi_aco_mo( ){ }
```

Comment:
@bluescarni 
If I uncomment this and I add the empty constructor, it gives a compiler error saying that the call to the constructor of pagmo::gi_aco_mo is ambiguous

Reply:
This happens because you already have a constructor which can be called without arguments, so when you write ``pagmo::gi_aco_mo algo``, the compiler does not know if it must call this constructor or the other one with all arguments defaulted.

As long as the algorithm has a constructor with all arguments defaulted, you can remove this other constructor.